packages/fetchai/connections/p2p_noise/connection.py

```diff
@@ -0,0 +1,431 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the p2p noise connection."""
+
+import asyncio
+import logging
+import os
+import subprocess
+import errno
+import struct
+import posix
+from pathlib import Path
+from publickey import PrivKey
+from typing import List, Optional, Sequence, Union
+
+from asyncio import AbstractEventLoop, CancelledError
+
+from aea.configurations.base import ConnectionConfig, PublicId
+from aea.connections.base import Connection
+from aea.mail.base import Address, Envelope
+
+logger = logging.getLogger(__name__)
+
+
+NOISE_NODE_SOURCE = os.path.join(
+    os.path.abspath(os.path.dirname(__file__)), "noise_node.go"
+)
+NOISE_NODE_CLARGS = list()
+
+NOISE_NODE_LOG_FILE = "noise_node.log"
+
+
+# TOFIX(LR) error: Cannot add child handler, the child watcher does not have a loop attached
+async def _async_golang_get_deps(src: str, loop: AbstractEventLoop):
+    cmd = ["go", "get", "-v", "-d", "."]
+
+    try:
+        logger.debug(cmd, loop)
+        proc = await asyncio.create_subprocess_exec(
+            *cmd, cwd=os.path.dirname(src), loop=loop
+        )
+    except Exception as e:
+        logger.error("While executing go get : {}".format(str(e)))
+        raise e
```

Comment:
a nice `TODO` is to put this into `aea.helpers`, because it might be useful for other code.

Reply:
also with the code below

Reply:
sure.
However, this function raise an exception when run with python3.6.
I already  have a TOFIX to address it.